opencog/atoms/base/Handle.cc

```diff
@@ -215,6 +215,30 @@ std::string hps_to_string(const HandlePairSeq& hps)
 	}
 	return ss.str();
 }
+std::string hc_to_string(const HandleCounter& hc)
+{
+	std::stringstream ss;
+	ss << "size = " << hc.size() << std::endl;
+	size_t i = 0;
+	for (const auto el : hc) {
+		ss << "atom[" << i << "]:" << std::endl << oc_to_string(el.first)
+		   << "num[" << i << "]:" << el.second << std::endl;
+		i++;
+	}
+	return ss.str();
+}
+std::string huc_to_string(const HandleUCounter& huc)
```

Comment:
do we really need both signed and unsigned versions? That strikes me as cheezy laziness. Pick one, pick the other, don't do both.

Reply:
anyway, we have count truth values for counting.  And other kinds of numeric values.  I bet that it probably makes more sense to use those, instead of using this.

Reply:
Regarding int vs unsigned int, I tend to use unsigned when I want to indicate to the developer the domain of the integer type more than anything else. I agree about count truth value, I might use that.